### ISSUE TITLE
hide impl classes in Navigate > Class

### ISSUE THREAD

**Issue by gavinking** [ceylon/ceylon-ide-intellij#327]
@bjansen would it be possible to eliminate the e.g. `Sequence$impl` classes in the result list?

**Comment by bjansen** [comment]
I can hide them, but then they won't be visible from Java files either. I think they might be useful when we try to implement a Ceylon interface in Java. I'll have to dig deeper.

**Comment by gavinking** [comment]
> I think they might be useful when we try to implement a Ceylon interface in Java.

Well, *perhaps*, but it's probably a bad idea to do this, especially since the $impl classes are going away, and for sure we don't do this anywhere right now, except in the language module which is a special case.

>  I'll have to dig deeper.

That's fine, but in the meantime, I still think we should just hide them.

**Comment by bjansen** [comment]
If we hide them, it means they're not in the index, which means the model loader won't be available to find them, which means `List$impl` won't be found, which means every piece of code that uses iterables will have (invalid) red highlights :(

**Comment by gavinking** [comment]
WDYM "code that uses iterable"? I don't understand.

**Comment by gavinking** [comment]
A compromise solution might be to add a `Weigher` so that the fall to the bottom of the list.

**Comment by bjansen** [comment]
> for sure we don't do this anywhere right now

Just found a place [where we do this](https://github.com/ceylon/ceylon-ide-intellij/blob/85b876092b99477eb26355c202074ea34403629b/plugin-ceylon-code/src/org/intellij/plugins/ceylon/ide/ceylonCode/psi/CeylonFile.java#L177-L181).

**Comment by davidfestal** [comment]
well, this code is temporary and will probably change. But it's right that Java code that implements a Ceylon interface with some implementation methods has to manage `$impl` elements explicitly.

**Comment by gavinking** [comment]
> Just found a place [where we do this](https://github.com/ceylon/ceylon-ide-intellij/blob/85b876092b99477eb26355c202074ea34403629b/plugin-ceylon-code/src/org/intellij/plugins/ceylon/ide/ceylonCode/psi/CeylonFile.java#L177-L181).

Then I suggest that we fix that code.

**Comment by gavinking** [comment]
> But it's right that Java code that implements a Ceylon interface with some implementation methods has to manage `$impl` elements explicitly. 

No. It's not right. `$impl`s are going away, and any code which uses them directly _will break_.

**Comment by bjansen** [comment]
Moving to 1.3.5 because it's more an inconvenience than a critical bug.
